src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -68,6 +70,14 @@ class AuthScreens extends React.Component {
             authEndpoint: `${CONFIG.EXPENSIFY.URL_API_ROOT}api?command=Push_Authenticate`,
         }).then(subscribeToReportCommentEvents);
 
+        // If we have a validation code in our URL ensure we validate the associated login
+        if (this.props.currentURL.includes(`/${ROUTES.VALIDATE_LOGIN}/`)) {
```

Comment:
Is there any reason why we couldn't use a specific page to handle the validation URLs? It seems like that is a better convention to follow (ie. each URL is handled by its own controller, allowing for a separation of concerns and keeping from having a single monolithic controller doing a bunch of different things.

Reply:
I actually suggested something similar [here](https://github.com/Expensify/Expensify.cash/pull/2231#issuecomment-813602884) and it was what I first went to when building this out. However, we don't actually need any sort of view for this logic so I wasn't sure if it required its own page.

Reply:
We could maybe a component that just returns an `ActivityIndicator` and does the redirect logic when it mounts?

Reply:
Yeah, I'd be fine with that. I think it makes all the route handling more intuitive and consistent, which is more important to me than having a component that renders very little (if anything at all).